core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -511,7 +511,7 @@ private PlanRoot doPlanQuery()
     private void planDistribution(PlanRoot plan)
     {
         // plan the execution on the active nodes
-        DistributedExecutionPlanner distributedPlanner = new DistributedExecutionPlanner(splitManager, metadata, dynamicFilterService);
+        DistributedExecutionPlanner distributedPlanner = new DistributedExecutionPlanner(splitManager, metadata, dynamicFilterService, new TypeAnalyzer(sqlParser, metadata));
```

Comment:
inject `TypeAnalyzer` from guice

Reply:
I was thinking about that but I noticed that in the `doPlanQuery` method it is created the same way so I thought it is better to be consisted. On the other hand I can replace both occurrences with an injected value. 
WDYT ?

Reply:
make sense to change both and not construct `TypeAnalyzer` manually.

Reply:
ok

Reply:
this should be a prep commit